federation/pkg/federation-controller/secret/secret_controller.go

```diff
@@ -55,34 +55,37 @@ var (
 	RequiredResources = []schema.GroupVersionResource{apiv1.SchemeGroupVersion.WithResource("secrets")}
 )
 
-type SecretController struct {
-	// For triggering single secret reconciliation. This is used when there is an
-	// add/update/delete operation on a secret in either federated API server or
-	// in some member of the federation.
-	secretDeliverer *util.DelayingDeliverer
-
-	// For triggering all secrets reconciliation. This is used when
-	// a new cluster becomes available.
+// FederationSyncController synchronizes the state of a federated type
+// to clusters that are members of the federation.
+type FederationSyncController struct {
```

Comment:
Apologies for the delay in checking this.
I have only one main comment. I see that this code has already gone into review and multiple revisions.
I believe u have made these changes within secret_controller.go itself for the people to appreciate the diff between the generic code and the existing controller code and some follow up PR will rename/move this to a generic named file. Right? 
The ```FederationSyncController {... }``` and the related code ideally should go into a common file, probably with unit test also moved suitably for the common code. I guess it will be better that this is merged in that shape when it is merged. (I am also quite fine, if this is merged as is and that change follows).
I personally like this overall and appreciate this contribution..
Additionally, I think @mwielgus might be interested in looking into this, with his wider understanding of controllers and informers, and since he self assigned this.